notebooks/metatranscriptomics_pipeline_iSeqnorm.ipynb

```diff
@@ -0,0 +1,1477 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_rna_concs_fp = './test_data/Quant/MiniPico/2024_NPH_Plates_3-6_initial_RNA_quant.txt'\n",
+    "\n",
+    "if not os.path.isfile(sample_rna_concs_fp):\n",
+    "    print(\"Problem! %s is not a path to a valid file\" % file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "sample_rna_concs = read_pico_csv(sample_rna_concs_fp, plate_reader='SpectraMax_i3x',conc_col_name='Sample RNA Concentration')\n",
+    "\n",
+    "plate_df = pd.merge(plate_df, sample_rna_concs, on='Well')\n",
+    "\n",
+    "well_col = 'Well'\n",
+    "\n",
+    "plate_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Visualize plate DNA/RNA concentrations and plate map:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get DNA & RNA concentratin information\n",
+    "dna_concs = make_2D_array(plate_df, data_col='Sample DNA Concentration', well_col=well_col).astype(float)\n",
+    "rna_concs = make_2D_array(plate_df, data_col='Sample RNA Concentration', well_col=well_col).astype(float)\n",
+    "\n",
+    "# get information for annotation\n",
+    "names = make_2D_array(plate_df, data_col='Sample', well_col=well_col)\n",
+    "\n",
+    "plot_plate_vals(dna_concs,\n",
+    "                annot_str=names,\n",
+    "                color_map='viridis',\n",
+    "                annot_fmt='.5s')\n",
+    "plt.title(\"DNA Concentrations\",fontsize=16)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Visualize plate RNA concentrations and plate map:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_plate_vals(rna_concs,\n",
+    "                annot_str=names,\n",
+    "                color_map='viridis',\n",
+    "                annot_fmt='.5s')\n",
+    "plt.title(\"RNA Concentrations\",fontsize=16)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sns.boxplot(x='Project Plate',y='Sample DNA Concentration',data=plate_df,fliersize=0)\n",
+    "sns.stripplot(x='Project Plate',y='Sample DNA Concentration',data=plate_df,color='grey',alpha=0.5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for plate in plate_df['Project Plate'].unique():\n",
+    "    plate_needs_dilution = requires_dilution(plate_df.loc[plate_df['Project Plate']==plate,\n",
+    "                          ['Sample DNA Concentration']], threshold=40)\n",
```

Comment:
A comment explaining what the hard-coded threshold=40 means would be helpful for maintenance in the future.

Also, I see that it is used in two places in this block.  If these two uses should always be the same, then it might be a good idea to put the threshold value in a variable and use the variable in both places, to make the value easier to change if needed.

Reply:
sounds good. Will include the comment explaining the threshold value and potentially define it in a variable